mobile/cmp/grid/impl/ColChooserModel.js

```diff
@@ -60,6 +70,35 @@ export class ColChooserModel {
         this.isOpen = false;
     }
 
+    updatePinnedColumn() {
+        const columns = [...this.columns];
+
+        // Loop through and, if applicable, pin the first
+        // non-excluded visible column encountered
+        let shouldPinFirst = this.pinFirst;
+        columns.forEach(it => {
+            if (it.exclude) return;
+            if (!it.hidden && shouldPinFirst) {
+                it.pinned = 'left';
+                shouldPinFirst = false;
+            } else {
+                it.pinned = null;
+            }
+        });
+
+        this.setColumns(columns);
+    }
+
+    onHideBtnClick(colId, hide) {
```

Comment:
I think that 'event handlers' like this on the model  have a bit of a code smell.  The Model should be independent of the actual affordances in the component, as much as possible.

Is there a method name that makes more sense for this here?

Reply:
Good catch Lee.

I moved the implementation of this handler to the component itself, where it leverages the `moveToIndex()`, `setHidden()` and `updatePinnedColumn()` methods on the model. This provides a nice symmetry with the `onDragEnd` event handler.